pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, opencv
+, opencv4
+, ceres-solver
+, suitesparse
+, metis
+, eigen
+, pkg-config
+, pybind11
+, numpy
+, pyyaml
+, lapack
+, gtest
+, gflags
+, glog
+, pytestCheckHook
+, networkx
+, pillow
+, exifread
+, gpxpy
+, pyproj
+, python-dateutil
+, joblib
+, repoze_lru
+, xmltodict
+, cloudpickle
+, scipy
+}:
+
+let
+  pname = "OpenSfM";
+  version = "0.5.1";
+  src = fetchFromGitHub {
+    owner = "mapillary";
+    repo = "OpenSfM";
+    rev = "v${version}";
+    sha256 = "sha256-AU3n3AVgNWeRBFt5hnuzDVU4WEWhI9mmKUuARnFBY4c=";
+    fetchSubmodules = true;
+  };
+  opensfm =
+    buildPythonPackage {
+      inherit pname version src;
+      nativeBuildInputs = [ cmake pkg-config ];
+      buildInputs = [
+        opencv
+        ceres-solver.dev
+        suitesparse
+        metis
+        eigen
+        lapack
+        gflags
+        gtest
+        glog
+      ];
+      propagatedBuildInputs = [
+        numpy
+        scipy
+        pyyaml
+        opencv4
+        networkx
+        pillow
+        exifread
+        gpxpy
+        pyproj
+        python-dateutil
+        joblib
+        repoze_lru
+        xmltodict
+        cloudpickle
+      ];
+      checkInputs = [ pytestCheckHook ];
+      dontUseCmakeBuildDir = true;
+      dontUseCmakeConfigure = true;
+      cmakeFlags = [
+        "-Bcmake_build"
+        "-Sopensfm/src"
+      ];
+      patches = [ ./fix-cmake.patch ];
+      postPatch = ''
+        rm opensfm/src/cmake/FindEigen.cmake
+        rm opensfm/src/cmake/FindCeres.cmake
+        rm opensfm/src/cmake/FindGlog.cmake
+        rm opensfm/src/cmake/FindGflags.cmake
+        rm -rf opensfm/src/third_party/gtest
+      '';
+      disabledTestPaths = [
+        "opensfm/test/test_pairs_selection.py"
+      ];
+      disabledTests = [
+        "test_run_all"
+      ];
+      outputs = [ "out" "viewer" ];
+      postInstall = ''
+        cp --no-preserve=mode -r ./viewer/ $viewer
```

Comment:
This should use moveToOutput.

Reply:
The change for c++ standard from c++11 to c++14 is already there, but fixing Eigen, etc - not, and I suspect it'd take a lot of time to get it there

I definitely will bring it up, but I'd rather start with the current approach

Reply:
@SuperSandro2000 I'm copying this path from the source directory, not from `$out`. I've tried `moveToOutput` and it only seems to handle paths in other outputs.

By the way, there seems to be no error handling in that function. When it didn't find any matches for the `viewer/` pattern it silently proceeded with the build and the first error I got was that the `$viewer` output was expected but hasn't been created by the build

Reply:
On the second thought, the derivation's user can always ask for `${opensfm.src}/viewer`, so I'm removing the $viewer output to keep things simple